Add Helm3 integration tests to CI

nrjpoddar opened the pull request:
Create new presubmit Prow check for running Helm3 integration tests

Commit:
Add Helm3 integration tests to CI

Commit:
Add make gen changes

howardjohn commented on prow/config/jobs/istio.yaml:
```diff
@@ -170,6 +170,11 @@ jobs:
         value: "-multicluster"
 
+  - name: integ-helm-tests
+    type: presubmit
```
We should add this as both presubmit and postsubmit to track flakes, etc like all of our other tests. and I recommend adding `modifiers: [hidden, optional]` for a while so we can ensure it stabilizes before we block every PR on it as we have not proven it is a stable test yet

nrjpoddar replied:
Makes sense, done

nrjpoddar replied:
I don't know how to add it for both pre & post submit. I deleted the type which based on README makes it run for both. What target should i use?

Commit:
Run Helm3 tests on both pre and post submit

Mark the test optional to prevent it from blocking PRs until the test
stabilizes.

Commit:
Add make gen output

Commit:
Run tests on both pre & post submit
